src/sentry/api/endpoints/organization_sentry_function_details.py

```diff
@@ -39,13 +39,18 @@ def put(self, request, organization, function):
             return Response(serializer.errors, status=400)
 
         data = serializer.validated_data
-        function.update(**data)
 
-        update_function(
-            function.code, function.external_id, data.get("overview", None), data["env_variables"]
-        )
-
-        return Response(serialize(function), status=201)
+        try:
+            update_function(
+                data.get("code"),
+                function.external_id,
+                data.get("overview", None),
+                data["env_variables"],
+            )
+            function.update(**data)
+            return Response(serialize(function), status=201)
+        except FailedPrecondition:
+            return Response({"detail": "Function is currently busy, try again later."}, status=400)
```

Comment:
You can use ParseError() from `rest_framework.exceptions` to do this with less 'code'.